pex/pex_info.py

```diff
@@ -123,6 +123,8 @@ def __init__(self, info=None):
                        '%s of type %s' % (info, type(info)))
     self._pex_info = info or {}
     self._distributions = self._pex_info.get('distributions', {})
+    # cast as set because pex info from json must store interpreter_constraints as a list
```

Comment:
Not sure I understand this comment, or maybe it's a typo. You're casting it as a set, but interpreter constraints needs to be stored as a list?

Reply:
Reading in a pex info using `from_pex` or `from_json` will instantiate a PexInfo object with json loaded as a dict. Interpreter constraints get serialized to json as a list b.c. json does not support Python sets, so casting it here is to cover the case where we read in a populated pex-info from json (which will be a list), however we want to store it internally as a set.